packages/core/src/indexing-store/cache.ts

```diff
@@ -379,32 +440,37 @@ export const createIndexingCache = ({
         .returning()
         .then((result) => result.length > 0);
     },
-    async flush({ db }) {
-      for (const table of cache.keys()) {
-        const batchSize = Math.round(
-          common.options.databaseMaxQueryParameters /
-            Object.keys(getTableColumns(table)).length,
-        );
+    async flush({ client }) {
+      let flushCount = 0;
+
+      const copy = getCopyHelper({ client });
 
+      for (const table of cache.keys()) {
         const tableSpillover = cache.get(table)!;
 
         const insertValues = Array.from(insertBuffer.get(table)!.values());
         const updateValues = Array.from(updateBuffer.get(table)!.values());
 
+        flushCount += insertValues.length + updateValues.length;
+
         if (insertValues.length > 0) {
-          for (const insertChunk of chunk(insertValues, batchSize)) {
-            await database.record(
-              { method: `${getTableName(table)}.flush()` },
-              async () => {
-                await db
-                  .insert(table)
-                  .values(insertChunk.map(({ row }) => row))
-                  .catch((error) => {
-                    throw parseSqlError(error);
-                  });
-              },
-            );
-          }
+          common.logger.debug({
+            service: "indexing",
+            msg: `Inserting ${insertValues.length} cached '${getTableName(table)}' rows into the database`,
+          });
+
+          const text = getCopyText(
+            table,
+            insertValues.map(({ row }) => row),
```

Comment:
Didn't we have to be careful last time about memory here, to make sure the insertValues are garbage-collectable as we build the text?

Reply:
Previously, yes. That was when we were handling a cache that was too large and flushing buffer entries to the database at the same step. Now because the buffer entries get flushed in smaller and more frequent chunks it isn't necessary.